buffer/src/main/java/io/netty/buffer/UnsafeByteBufUtil.java

```diff
@@ -544,23 +544,23 @@ static void setBytes(AbstractByteBuf buf, long addr, int index, byte[] src, int
         }
     }
 
-    static void setBytes(AbstractByteBuf buf, long addr, int index, ByteBuffer src) {
+    static void setBytes(AbstractByteBuf buf, MemoryAddress addr, int index, ByteBuffer src) {
         final int length = src.remaining();
         if (length == 0) {
             return;
         }
 
-        if (src.isDirect()) {
-            buf.checkIndex(index, length);
-            // Copy from direct memory
-            long srcAddress = PlatformDependent.directBufferAddress(src);
-            PlatformDependent.copyMemory(srcAddress + src.position(), addr, length);
-            src.position(src.position() + length);
-        } else if (src.hasArray()) {
+        if (src.hasArray()) {
             buf.checkIndex(index, length);
             // Copy from array
             PlatformDependent.copyMemory(src.array(), src.arrayOffset() + src.position(), addr, length);
             src.position(src.position() + length);
+        } else if (src.isDirect()) {
+            buf.checkIndex(index, length);
+            // Copy from direct memory
+            MemoryAddress srcAddress = PlatformDependent.directBufferAddress(src);
```

Comment:
I might be wrong but can you not call src.MemoryAddress() on this to get the addr.

Reply:
Good shout, I've changed it

Reply:
I mistook src for a `Buffer` (which has the `getBufferAddress()` method) when it is instead a `ByteBuffer` - so the `PlatformDependent.directBufferAddress(src)` has remained